app/controllers/source_data_controller.rb

```diff
@@ -86,4 +95,14 @@ def show
     @klass.joins(:data_source).
       merge(GrdaWarehouse::DataSource.editable_by(current_user).source)
   end
+
+  private def hmis_url
+    return unless @data_source.hmis
+
+    base = "https://#{@data_source.hmis}"
+    return "#{base}/projects/#{@item.id}" if @item.is_a?(GrdaWarehouse::Hud::Project)
+    return "#{base}/organizations/#{@item.id}" if @item.is_a?(GrdaWarehouse::Hud::Organization)
+    return "#{base}/client/#{@item.id}" if @item.is_a?(GrdaWarehouse::Hud::Client)
+    return "#{base}/client/#{@item.client.id}/enrollments/#{@item.enrollment.id}" if @item.respond_to?(:enrollment) && @item.respond_to?(:client)
+  end
```

Comment:
This is probably fine for now, but I wonder if we should delegate these urls to the data source, or some more generic class, so we can use them around the app.